driver-async/src/test/functional/com/mongodb/async/client/NettyStreamFactoryFactorySpecification.groovy

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.async.client
+
+import com.mongodb.async.FutureResultCallback
+import com.mongodb.connection.netty.NettyStreamFactoryFactory
+import io.netty.channel.oio.OioEventLoopGroup
+import io.netty.channel.socket.oio.OioSocketChannel
+import org.bson.Document
+
+import static com.mongodb.async.client.Fixture.getMongoClientBuilderFromConnectionString
+import static java.util.concurrent.TimeUnit.SECONDS
+
+class NettyStreamFactoryFactorySpecification extends FunctionalSpecification {
+
+    private MongoClient mongoClient
+
+    def 'should allow a custom Event Loop Group and Socket Channel'() {
```

Comment:
This test is too indirect to assert what we're trying to assert.

What about instead adding getter methods to NettyStream for all the configurable properties, and just assert that they take on the values configured in the FactoryFactory?